lib/src/manup_service.dart

```diff
@@ -65,13 +71,89 @@ class ManUpService {
   @visibleForTesting
   Future<Metadata> getMetadata() async {
     try {
-      var data = await this.http.get(this.url);
-      this.http.close();
-      
+      var client = this.delegate.httpClient;
+      var data = await client.get(this.url);
+      client.close();
       Map<String, dynamic> json = jsonDecode(data.body);
       return Metadata(data: json);
     } catch (exception) {
       throw ManUpException(exception.toString());
     }
   }
+
+  // manup status validation
+  _handleManUpStatus(ManUpStatus status) {
+    switch (status) {
+      case ManUpStatus.latest:
+        this.delegate?.manUpFinishedValidation?.call();
+        return;
+      case ManUpStatus.supported:
+        this.delegate?.manUpUpdateAvailable?.call();
+        break;
+      case ManUpStatus.unsupported:
+        this.delegate?.manUpUpdateRequired?.call();
+        break;
+      case ManUpStatus.disabled:
+        this.delegate?.manUpMaintenanceMode?.call();
+        break;
+      default:
+        return;
+    }
+    BuildContext context = this?.delegate?.appContext;
+    if (context != null && this?.delegate?.shouldShowManupAlert == true) {
+      showManupDialog(context, status, this).then((isDone) =>
```

Comment:
To me passing a `this` instance to a mixin method seems a bit confusing. I'd probably rather see it just pass in the actual data it needs (i.e. the metadata and the updateUrl.

Reply:
Service itself is required to call the retry function.

Reply:
You can pass that in as an argument as well:

```dart
showManupDialog(() => retry());

// ...
Future<bool> showManUpDialog(
    Function retry, /* ... */
)
```

Reply:
I'm not sure that ManUp should behave this way, as IIRC in the iOS build you would only show the dialog with no actions: https://github.com/NextFaze/ManUp/blob/master/ManUp/ManUp.m#L299

The ManUp service should be running each time the app is brought back into the foreground, with checks on either time since last update (Current iOS way) or checking for etag/modified times for the files

Reply:
just doing that 

 ```
 Future<bool> _launchUrl() {
    var launchUrl = configData.updateUrl;
    return canLaunch(launchUrl)
        .then((canLaunch) => canLaunch ? launch(launchUrl) : false);
  }

  Future<ManUpStatus> _retry() {
    return this.validate();
  }
```